vendor/src/github.com/docker/libnetwork/endpoint_cnt.go

```diff
@@ -108,6 +108,25 @@ func (ec *endpointCnt) EndpointCnt() uint64 {
 	return ec.Count
 }
 
+func (ec *endpointCnt) updateStore() error {
+retry:
```

Comment:
No timeout or retry limit is needed. The only case the retry path is taken is the legitimate case when the store rejects our update because the store object has since changed, because somebody else modified in between.
In such case this api pulls from the store the latest version of the object and re-perform the computation.

This is a common pattern in libnetwork when interacting with the datastore.